velox/expression/ExprRewriteRegistry.cpp

```diff
@@ -28,6 +28,16 @@ void ExprRewriteRegistry::clear() {
 
 core::TypedExprPtr ExprRewriteRegistry::rewrite(
     const core::TypedExprPtr& expr) {
+  const auto& inputs = expr->inputs();
+  VELOX_CHECK(
```

Comment:
We may need to relax this check later if we get any re-writes for non-deterministic expressions.